prepare_iso/prepare_vdi.sh

```diff
@@ -0,0 +1,273 @@
+
+
+msg_status() {
+	echo "\033[0;32m-- $1\033[0m"
+}
+msg_error() {
+	echo "\033[0;31m-- $1\033[0m"
+}
+
+render_template() {
+	eval "echo \"$(cat "$1")\""
+}
+
+if [ $# -eq 0 ]; then
+	usage
+	exit 1
+fi
+
+SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd)"
+SUPPORT_DIR="$SCRIPT_DIR/support"
+
+# Parse the optional command line switches
+USER="vagrant"
+PASSWORD="vagrant"
+IMAGE_PATH="$SUPPORT_DIR/vagrant.jpg"
+
+# Flags
+DISABLE_REMOTE_MANAGEMENT=0
+DISABLE_SCREEN_SHARING=0
+DISABLE_SIP=0
+
+while getopts u:p:i:o:D: OPT; do
+  case "$OPT" in
+    u)
+      USER="$OPTARG"
+      ;;
+    p)
+      PASSWORD="$OPTARG"
+      ;;
+    i)
+      IMAGE_PATH="$OPTARG"
+      ;;
+    o)
+      OUTPUT_DMG="$OPTARG"
+      ;;
+    D)
+      if [ x${!OPTARG} = x0 ]; then
+        eval $OPTARG=1
+      elif [ x${!OPTARG} != x1 ]; then
+        msg_error "Unknown flag: ${OPTARG}"
+        usage
+        exit 1
+      fi
+      ;;
+    \?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+# Remove the switches we parsed above.
+shift $(expr $OPTIND - 1)
+
+if [ $(id -u) -ne 0 ]; then
+	msg_error "This script must be run as root, as it saves a disk image with ownerships enabled."
+	exit 1
+fi
+
+ESD="$1"
+if [ ! -e "$ESD" ]; then
+	msg_error "Input installer image $ESD could not be found! Exiting.."
+	exit 1
+fi
+
+if [ -d "$ESD" ]; then
+	# we might be an install .app
+	if [ -e "$ESD/Contents/SharedSupport/InstallESD.dmg" ]; then
+		ESD="$ESD/Contents/SharedSupport/InstallESD.dmg"
+	else
+		msg_error "Can't locate an InstallESD.dmg in this source location $ESD!"
+	fi
+fi
+
+VEEWEE_DIR="$(cd "$SCRIPT_DIR/../../../"; pwd)"
+VEEWEE_UID=$(/usr/bin/stat -f %u "$VEEWEE_DIR")
+VEEWEE_GID=$(/usr/bin/stat -f %g "$VEEWEE_DIR")
+DEFINITION_DIR="$(cd "$SCRIPT_DIR/.."; pwd)"
+
+if [ "$2" = "" ]; then
+    msg_error "Currently an explicit output directory is required as the second argument."
+	exit 1
+	# The rest is left over from the old prepare_veewee_iso.sh script. Not sure if we
+    # should leave in this functionality to automatically locate the veewee directory.
+	DEFAULT_ISO_DIR=1
+	OLDPWD=$(pwd)
+	cd "$SCRIPT_DIR"
+	# default to the veewee/iso directory
+	if [ ! -d "../../../iso" ]; then
+		mkdir "../../../iso"
+		chown $VEEWEE_UID:$VEEWEE_GID "../../../iso"
+	fi
+	OUT_DIR="$(cd "$SCRIPT_DIR"; cd ../../../iso; pwd)"
+	cd "$OLDPWD" # Rest of script depends on being in the working directory if we were passed relative paths
+else
+	OUT_DIR="$2"
+fi
+
+if [ ! -d "$OUT_DIR" ]; then
+	msg_status "Destination dir $OUT_DIR doesn't exist, creating.."
+	mkdir -p "$OUT_DIR"
+fi
+
+MNT_ESD=$(/usr/bin/mktemp -d /tmp/veewee-osx-esd.XXXX)
+msg_status "Attaching input OS X installer image"
+hdiutil attach "$ESD" -mountpoint "$MNT_ESD" -nobrowse -owners on
+if [ $? -ne 0 ]; then
+	[ ! -e "$ESD" ] && msg_error "Could not find $ESD in $(pwd)"
+	msg_error "Could not mount $ESD on $MNT_ESD"
+	exit 1
+fi
+
+msg_status "Mounting BaseSystem.."
+BASE_SYSTEM_DMG="$MNT_ESD/BaseSystem.dmg"
```

Comment:
In High Sierra BaseSystem.dmg is not inside the InstallESD.dmg, it is in the same directory, in `Installer.app/Contents/SharedSupport/BaseSystem.dmg`. You could check for the existence of the file in both paths, then you would have compatibility with High Sierra as well.